workers/unity/Assets/Gdk/Core/Components/Option.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+
+namespace Improbable.Gdk.Core
+{
+    public struct Option<T> : IEquatable<Option<T>>
+    {
+        private bool hasValue;
+        private T value;
+        
+        public Option(T newValue)
+        {
+            hasValue = false;
+            value = default(T);
+            Value = newValue;
+        }
+
+        public BlittableBool HasValue => hasValue;
+
+        public T Value
+        {
+            get
+            {
+                if (!hasValue)
+                {
+                    throw new InvalidOperationException("Called Value on empty Option.");
+                }
+
+                return value;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentException("newValue is null.");
```

Comment:
Please make a new exception type for this.

Reply:
How granular do you want this? as in do you want
- an OptionArgumentException
- an ArgumentIsNullException
- an OptionArgumentIsNullException?

Reply:
There is already a `System.ArgumentNullException` that should be used. https://msdn.microsoft.com/en-us/library/system.argumentnullexception(v=vs.110).aspx

Reply:
Also, is this really necessary? What's wrong with the standard exceptions?

Reply:
Is exception has been removed as Options are immutable now.